### PR TITLE
Adding configuration variable replacement in c_cpp_properties.json.

### PR DESCRIPTION
**Pull request by thejcannon**
As requested in issue #314 in addition to "env.ENV_VAR", "env:ENV_VAR" and just "ENV_VAR", adding support for "config:config_variable" and "config.config_variable".

### CODE REVIEW

File: Extension/src/common.ts
Comment (bobbrow) [inline Extension/src/common.ts]: >varType = "env"; } [](start = 37, length = 18)

I believe our linting rules require the body of the if to be on a new line.

Comment (thejcannon) [reply Extension/src/common.ts]: I ran TSLint and got "vscode-tslint: Status is OK" (It did find some other lint issues though).
Happy to move it to new line, though!

**Review by bobbrow — APPROVED** [review]
:shipit:

**Comment by bobbrow** [comment]
@thejcannon, I'm ready to accept this pull request.  You just need to sign the [CLA](https://cla.opensource.microsoft.com/Microsoft/vscode-cpptools?pullRequest=1529) if you haven't already then I can click the button.

Thanks for the help!

**Comment by thejcannon** [comment]
@bobbrow Just waiting on approval from my employer.

**Comment by thejcannon** [comment]
@bobbrow done!

**Comment by bobbrow** [comment]
Awesome.  Thanks again!

**Pull request merged**
